renderdoc/driver/vulkan/vk_shader_cache.cpp

```diff
@@ -1056,11 +1056,28 @@ void VulkanShaderCache::MakeGraphicsPipelineInfo(VkGraphicsPipelineCreateInfo &p
     rs.pNext = &provokeSetup;
   }
 
+  uint64_t flags = pipeInfo.flags;
   // never create derivatives
-  ret.flags &= ~VK_PIPELINE_CREATE_DERIVATIVE_BIT;
+  flags &= ~VK_PIPELINE_CREATE_DERIVATIVE_BIT;
 
-  ret.flags &= ~VK_PIPELINE_CREATE_LIBRARY_BIT_KHR;
-  ret.flags &= ~VK_PIPELINE_CREATE_RETAIN_LINK_TIME_OPTIMIZATION_INFO_BIT_EXT;
+  flags &= ~VK_PIPELINE_CREATE_LIBRARY_BIT_KHR;
+  flags &= ~VK_PIPELINE_CREATE_RETAIN_LINK_TIME_OPTIMIZATION_INFO_BIT_EXT;
+
+  static VkPipelineCreateFlags2CreateInfo createFlags = {
+      VK_STRUCTURE_TYPE_PIPELINE_CREATE_FLAGS_2_CREATE_INFO,
+  };
+
+  if(pipeInfo.useCreateFlags2 && m_pDriver->GetExtensions(GetRecord(m_Device)).ext_KHR_maintenance5)
```

Comment:
There are a couple of places that are checking just if the extension is enabled, but we should be checking for the extension feature to be enabled - see for example `WrappedVulkan::m_FragmentShadingRate` and `WrappedVulkan::FragmentShadingRate()`.

You can have the extension without the feature (degenerate if enabled as an extension itself but could happen implicitly from a core version), but you can't have the feature without the extension (either explicitly or implicitly).